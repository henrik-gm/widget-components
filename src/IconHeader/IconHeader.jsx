import React from 'react';
import PropTypes from 'prop-types';
import styles from './IconHeader.scss';

const IconHeader = ({ title, subtitle, children }) => (
   <header className={`KambiWidget-card-text-color KambiWidget-card-border-color KambiWidget-font ${styles.header}`}>
      <div className={styles.icon}>{children}</div>
      <div className={styles.container}>
         <div className={styles.title} title={title}>{title}</div>
         <div className={styles.subtitle} title={subtitle}>{subtitle}</div>
      </div>
   </header>
);

/**
 * @property [children] {ReactElement} Header's icon markup
 * @property title {string} Header's title
 * @property [subtitle] {string} Header's subtitle
 */
IconHeader.propTypes = {
   children: PropTypes.node,
   title: PropTypes.string.isRequired,
   subtitle: PropTypes.string
};

export default IconHeader;
