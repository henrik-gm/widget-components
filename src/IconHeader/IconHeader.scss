@mixin calc($property, $expression) {
   #{$property}: -moz-calc(#{$expression});
   #{$property}: -webkit-calc(#{$expression});
   #{$property}: calc(#{$expression});
}

@mixin inherited {
   border-bottom: solid 1px rgba(0, 0, 0, 0.05);
   font-size: 12px;
   line-height: 36px;
   min-height: 37px;
   overflow: hidden;
   position: relative;
}

@mixin size {
   color: black;
   line-height: 16px;
   max-height: 16px;
   min-height: 16px;
}

:local(.header) {
   @include inherited();
   display: flex;
   padding-left: 16px;
   align-items: center;
}

:local(.logo) {
   background-repeat: no-repeat;
   background-size: cover;
   height: 32px;
   margin-right: 12px;
   width: 32px;
}

:local(.container) {
   @include calc(max-width, '100% - 62px');
}

:local(.title) {
   @include size();
   font-size: 16px;
   font-weight: bold;
   text-transform: uppercase;
}

:local(.subtitle) {
   @include size();
}
