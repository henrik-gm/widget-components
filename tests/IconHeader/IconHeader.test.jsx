import React, { Children } from 'react';
import IconHeader from '../../src/IconHeader/IconHeader';
import ReactTestRenderer from 'react-test-renderer';

describe('IconHeader DOM rendering', () => {

   it('renders correctly with title', () => {
      const tree = ReactTestRenderer.create(
         <IconHeader title="test" />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with title and subtitle', () => {
      const tree = ReactTestRenderer.create(
         <IconHeader title="test" subtitle="Test subtitle" />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with inner element', () => {
      const tree = ReactTestRenderer.create(
         <IconHeader title="test">
            <div>Icon</div>
         </IconHeader>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});
