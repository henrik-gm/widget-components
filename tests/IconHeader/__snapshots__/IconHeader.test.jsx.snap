exports[`IconHeader DOM rendering renders correctly with inner element 1`] = `
<header
  className="KambiWidget-card-border-color KambiWidget-font header">
  <div
    className="icon">
    <div>
      Icon
    </div>
  </div>
  <div
    className="container">
    <div
      className="title">
      test
    </div>
    <div
      className="subtitle" />
  </div>
</header>
`;

exports[`IconHeader DOM rendering renders correctly with title 1`] = `
<header
  className="KambiWidget-card-border-color KambiWidget-font header">
  <div
    className="icon" />
  <div
    className="container">
    <div
      className="title">
      test
    </div>
    <div
      className="subtitle" />
  </div>
</header>
`;

exports[`IconHeader DOM rendering renders correctly with title and subtitle 1`] = `
<header
  className="KambiWidget-card-border-color KambiWidget-font header">
  <div
    className="icon" />
  <div
    className="container">
    <div
      className="title">
      test
    </div>
    <div
      className="subtitle">
      Test subtitle
    </div>
  </div>
</header>
`;
